File: C-Plus-Plus/Bit--Manipulation/Power_of_2.cpp
/*
Description : 
        We have to find wether the number is power of 2 or not . If it is then return yes else no 
*/

#include<bits/stdc++.h>
using namespace std;

bool power_of_2(int n)
{
	// if zero then it will return true
	return (n&& !(n&n-1));
}
int main()
{
	int num;
    cout<<"Enter the number : "<<endl;
	cin>>num;
	power_of_2(num)? cout<<"Yes\n": cout<<"No\n";
}

/*
Time complexity : O(1)
Space complexity : O(1)
*/

/*
Input :
Enter the number : 
64

Output :
Yes

*/

File: Julia/Bit--Manipulation/power_of_2.jl
# Julia program to check if a number is a power of two.

function power_of_two(n)
    # 0 is not a power of 2
    if(n != 0)
        if((n & (n-1)) == false)
            return true
        end
    end
    return false
end


print("Enter a number: ")
n = readline()
n = parse(Int, n)
res = power_of_two(n)
if(res)
    print("The given number is a power of 2.")
else
    print("The given number is not a power of 2.")
end


"""
Time Complexity - O(1)
Space Complexity - O(1)

SAMPLE INPUT AND OUTPUT

SAMPLE I
Enter a number: 4
The given number is a power of 2.

SAMPLE II
Enter a number: 0
The given number is not a power of 2.
"""
